src/main/ruby/core/truffle/boot.rb

```diff
@@ -16,9 +16,18 @@ def self.main_s
   end
 
   def self.check_syntax
-    inner_check_syntax
-    STDOUT.puts 'Syntax OK'
-    true
+    check_syntax_file(original_input_file) && ARGV.all? { |arg| check_syntax_file arg }
```

Comment:
Interestingly, this seems some non-standard behavior inherited from JRuby.
MRI only checks the main script:
```
ruby -v -c src/main/ruby/core/float.rb spec/ruby/command_line/fixtures/bad_syntax.rb
ruby 2.3.4p301 (2017-03-30 revision 58214) [x86_64-linux]
Syntax OK
```